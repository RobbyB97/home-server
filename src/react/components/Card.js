import React from 'react'
import {Link} from 'react-router-dom'

const Card = ({image, title, subtext, link}) => (
    <div>
        <div className="card__wrap">
            <a className="card__link" href={link}>
                <div className="card">
                    <img src={image}/>
                    <div className="card__text">
                        <h2>{title}</h2>
                        <p>{subtext}</p>
                    </div>
                </div>
            </a>
        </div>
    </div>
)

export default Card