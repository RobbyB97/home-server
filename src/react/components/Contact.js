import React from 'react';


export class Contact extends React.Component {
    render() {
        return (
            <section id="Contact" className="contact">
                <div className="contact__shape one"></div>
                <div className="contact__shape two"></div>
                Contact component
            </section>
        );
    }
}


export default Contact;