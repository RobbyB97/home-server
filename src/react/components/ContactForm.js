/* Libraries */
import React from 'react'


export class ContactForm extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            nameValue: ''
        };

        this.handleNameChange = this.handleNameChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleNameChange(event) {
        this.setState({
            ...this.state,
            nameValue: event.target.value
        });
    }

    handleSubmit(event) {
        console.log('hi')
    }

    render() {
        return (
            <div className="ContactForm">
                <br /><br /><br /><br /><br />
                <form id="contactForm">
                    <p>
                        <label>Name</label>
                        <input type="text" 
                            value={this.state.nameValue} 
                            onChange={this.handleNameChange} 
                            required 
                        />
                    </p>
                </form>
            </div>
        );
    }
}


export default ContactForm