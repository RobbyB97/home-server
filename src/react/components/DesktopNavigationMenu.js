import React from 'react'
import {NavLink} from 'react-router-dom'
import { connect } from 'react-redux'

import {setTitle} from '../actions/title'


const DesktopNavigationMenu = ({dispatch}) => (
    <div id="desktop-navigation-menu">
        <ul>
            <li>
                <NavLink to="/react/" onClick={() => {
                    dispatch(setTitle(''))
                }}>Home</NavLink>
            </li>
            <li>
                <NavLink to="/react/style_guide" onClick={() => {
                    dispatch(setTitle('Style Guide'))
                }}>Style Guide</NavLink>
            </li>
        </ul>
    </div>
)

export default connect()(DesktopNavigationMenu)