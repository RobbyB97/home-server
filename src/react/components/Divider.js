import React from 'react';


export class Divider extends React.Component {
    constructor(props) {
        super(props);
    }


    render() {
        return (
            <div className={`divider ${this.props.section}`}>
                <div className="divider__shape one"></div>
                <div className="divider__shape two"></div>
            </div>
        );
    }
}


export default Divider;