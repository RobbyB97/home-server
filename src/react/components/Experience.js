import React from 'react';


export class Experience extends React.Component {
    render() {
        return (
            <section id="Experience" className="experience">
                <div className="experience__shape one"></div>
                <div className="experience__shape two"></div>
                Experience component
            </section>
        );
    }
}


export default Experience;