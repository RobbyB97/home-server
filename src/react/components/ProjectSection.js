import React from 'react';


export class ProjectSection extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            active: true
        };
    }


    toggle = () => {
        this.setState({
            active: !this.state.active
        });
    };


    render() {
        return (
            <article className="projectSection">
                <header 
                    className="projectSection__header"
                    onClick={this.toggle}
                >
                    {this.props.logo &&
                        <img
                            className="projectSection__logo"
                            src={this.props.logo}
                        />
                    }

                    <h2 className="projectSection__name">
                        {this.props.name}
                    </h2>
                </header>

                <section
                    className="projectSection__description"
                    data-active={this.state.active}
                >
                    <div 
                        className="projectSection__languages"
                        onClick={this.toggle}    
                    >
                        {this.props.languages &&
                            this.props.languages.map((language) => (
                                <h5 key={language}>{language}</h5>
                            ))
                        }
                    </div>

                    <nav className="projectSection__links">
                        {this.props.github && 
                            <a 
                                className="projectSection__github"
                                href={this.props.github}
                                target="_blank"
                                rel="noopener"
                            >
                                <img src="/dist/images/icons/github/github-white-100.png"/>
                                <p>Github</p>
                            </a>
                        }

                        {this.props.website &&
                            <a 
                                className="projectSection__website"
                                href={this.props.website}
                                target="_blank"
                                rel="noopener"
                            >
                                Website
                            </a>
                        }

                        {this.props.customLinks &&
                            this.props.customLinks.map((link) => ((
                                <a 
                                    className="projectSection__customLink"
                                    href={link.link}
                                    target="_blank"
                                    rel="noopener"
                                    key={link.link}
                                >
                                        {link.text}
                                </a>
                            )))
                        }
                    </nav>

                    <summary 
                        className="projectSection__summary"
                        onClick={this.toggle}
                    >
                        {this.props.description}
                    </summary>
                </section>

                <footer
                    className="projectSection__footer"
                    data-active={this.state.active}
                    onClick={this.toggle}
                >
                    <p data-active={this.state.active}>^</p>
                </footer>
            </article>
        );
    }
}


export default ProjectSection;