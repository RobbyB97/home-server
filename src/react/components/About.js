import React from 'react';

import Divider from './Divider'


export class About extends React.Component {
    render() {
        return (
            <section id="About" className="about">

                <div className="about__content one">
                    <div className="about__portrait">
                        <img src="/dist/images/aboutImage.png" />
                        <div className="about__text one">
                            <p>
                                <span className="one">Hi, my name's Robby. <span className="two">I'm a front-end developer and digital artist</span>.</span>
                            </p>
                        </div>
                    </div>
                </div>

                <Divider section="about two" />

                <div className="about__content two">
                    <div className="about__text two">
                        <p>
                            I want to help you succeed by building a beautiful, fast and functional website for your business.
                        </p>
                    </div>
                </div>
            </section>
        );
    }
}


export default About;