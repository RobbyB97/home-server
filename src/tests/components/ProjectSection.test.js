import React from 'react';
import { shallow } from 'enzyme';

import { ProjectSection } from '../../react/components/ProjectSection';
import projectSections from '../fixtures/projectSections';


test('Render ProjectSection without props', () => {
    const wrapper = shallow(<ProjectSection />);
    expect(wrapper).toMatchSnapshot();
});

test('Render ProjectSection with logo', () => {
    const wrapper = shallow(<ProjectSection {...projectSections[0]}/>);
    expect(wrapper).toMatchSnapshot();
});

test('Render ProjectSection with name', () => {
    const wrapper = shallow(<ProjectSection {...projectSections[1]}/>);
    expect(wrapper).toMatchSnapshot();
});

test('Render ProjectSection with languages', () => {
    const wrapper = shallow(<ProjectSection {...projectSections[2]}/>);
    expect(wrapper).toMatchSnapshot();
});

test('Render ProjectSection with Github link', () => {
    const wrapper = shallow(<ProjectSection {...projectSections[3]}/>);
    expect(wrapper).toMatchSnapshot();
});

test('Render ProjectSection with website link', () => {
    const wrapper = shallow(<ProjectSection {...projectSections[4]}/>);
    expect(wrapper).toMatchSnapshot();
});

test('Render ProjectSection with custom links', () => {
    const wrapper = shallow(<ProjectSection {...projectSections[5]}/>);
    expect(wrapper).toMatchSnapshot();
});

test('Render ProjectSection with description', () => {
    const wrapper = shallow(<ProjectSection {...projectSections[6]}/>);
    expect(wrapper).toMatchSnapshot();
});

test('Render ProjectSection with all props', () => {
    const wrapper = shallow(<ProjectSection {...projectSections[7]}/>);
    expect(wrapper).toMatchSnapshot();
});