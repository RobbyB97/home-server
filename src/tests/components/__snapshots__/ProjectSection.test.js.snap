// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ProjectSection with Github link 1`] = `
<article
  className="projectSection"
>
  <header
    className="projectSection__header"
    onClick={[Function]}
  >
    <h2
      className="projectSection__name"
    />
  </header>
  <section
    className="projectSection__description"
    data-active={true}
  >
    <div
      className="projectSection__languages"
      onClick={[Function]}
    />
    <nav
      className="projectSection__links"
    >
      <a
        className="projectSection__github"
        href="Github!"
        rel="noopener"
        target="_blank"
      >
        <img
          src="/dist/images/icons/github/github-white-100.png"
        />
        <p>
          Github
        </p>
      </a>
    </nav>
    <summary
      className="projectSection__summary"
      onClick={[Function]}
    />
  </section>
  <footer
    className="projectSection__footer"
    data-active={true}
    onClick={[Function]}
  >
    <p
      data-active={true}
    >
      ^
    </p>
  </footer>
</article>
`;

exports[`Render ProjectSection with all props 1`] = `
<article
  className="projectSection"
>
  <header
    className="projectSection__header"
    onClick={[Function]}
  >
    <img
      className="projectSection__logo"
      src="Logo!"
    />
    <h2
      className="projectSection__name"
    >
      Name!
    </h2>
  </header>
  <section
    className="projectSection__description"
    data-active={true}
  >
    <div
      className="projectSection__languages"
      onClick={[Function]}
    >
      <h5
        key="Language1!"
      >
        Language1!
      </h5>
      <h5
        key="Language2!"
      >
        Language2!
      </h5>
    </div>
    <nav
      className="projectSection__links"
    >
      <a
        className="projectSection__github"
        href="Github!"
        rel="noopener"
        target="_blank"
      >
        <img
          src="/dist/images/icons/github/github-white-100.png"
        />
        <p>
          Github
        </p>
      </a>
      <a
        className="projectSection__website"
        href="Website!"
        rel="noopener"
        target="_blank"
      >
        Website
      </a>
      <a
        className="projectSection__customLink"
        href="Custom link link!"
        key="Custom link link!"
        rel="noopener"
        target="_blank"
      />
    </nav>
    <summary
      className="projectSection__summary"
      onClick={[Function]}
    >
      Description!
    </summary>
  </section>
  <footer
    className="projectSection__footer"
    data-active={true}
    onClick={[Function]}
  >
    <p
      data-active={true}
    >
      ^
    </p>
  </footer>
</article>
`;

exports[`Render ProjectSection with custom links 1`] = `
<article
  className="projectSection"
>
  <header
    className="projectSection__header"
    onClick={[Function]}
  >
    <h2
      className="projectSection__name"
    />
  </header>
  <section
    className="projectSection__description"
    data-active={true}
  >
    <div
      className="projectSection__languages"
      onClick={[Function]}
    />
    <nav
      className="projectSection__links"
    >
      <a
        className="projectSection__customLink"
        href="Custom link link!"
        key="Custom link link!"
        rel="noopener"
        target="_blank"
      />
    </nav>
    <summary
      className="projectSection__summary"
      onClick={[Function]}
    />
  </section>
  <footer
    className="projectSection__footer"
    data-active={true}
    onClick={[Function]}
  >
    <p
      data-active={true}
    >
      ^
    </p>
  </footer>
</article>
`;

exports[`Render ProjectSection with description 1`] = `
<article
  className="projectSection"
>
  <header
    className="projectSection__header"
    onClick={[Function]}
  >
    <h2
      className="projectSection__name"
    />
  </header>
  <section
    className="projectSection__description"
    data-active={true}
  >
    <div
      className="projectSection__languages"
      onClick={[Function]}
    />
    <nav
      className="projectSection__links"
    />
    <summary
      className="projectSection__summary"
      onClick={[Function]}
    >
      Description!
    </summary>
  </section>
  <footer
    className="projectSection__footer"
    data-active={true}
    onClick={[Function]}
  >
    <p
      data-active={true}
    >
      ^
    </p>
  </footer>
</article>
`;

exports[`Render ProjectSection with languages 1`] = `
<article
  className="projectSection"
>
  <header
    className="projectSection__header"
    onClick={[Function]}
  >
    <h2
      className="projectSection__name"
    />
  </header>
  <section
    className="projectSection__description"
    data-active={true}
  >
    <div
      className="projectSection__languages"
      onClick={[Function]}
    >
      <h5
        key="Language1!"
      >
        Language1!
      </h5>
      <h5
        key="Language2!"
      >
        Language2!
      </h5>
    </div>
    <nav
      className="projectSection__links"
    />
    <summary
      className="projectSection__summary"
      onClick={[Function]}
    />
  </section>
  <footer
    className="projectSection__footer"
    data-active={true}
    onClick={[Function]}
  >
    <p
      data-active={true}
    >
      ^
    </p>
  </footer>
</article>
`;

exports[`Render ProjectSection with logo 1`] = `
<article
  className="projectSection"
>
  <header
    className="projectSection__header"
    onClick={[Function]}
  >
    <img
      className="projectSection__logo"
      src="Logo!"
    />
    <h2
      className="projectSection__name"
    />
  </header>
  <section
    className="projectSection__description"
    data-active={true}
  >
    <div
      className="projectSection__languages"
      onClick={[Function]}
    />
    <nav
      className="projectSection__links"
    />
    <summary
      className="projectSection__summary"
      onClick={[Function]}
    />
  </section>
  <footer
    className="projectSection__footer"
    data-active={true}
    onClick={[Function]}
  >
    <p
      data-active={true}
    >
      ^
    </p>
  </footer>
</article>
`;

exports[`Render ProjectSection with name 1`] = `
<article
  className="projectSection"
>
  <header
    className="projectSection__header"
    onClick={[Function]}
  >
    <h2
      className="projectSection__name"
    >
      Name!
    </h2>
  </header>
  <section
    className="projectSection__description"
    data-active={true}
  >
    <div
      className="projectSection__languages"
      onClick={[Function]}
    />
    <nav
      className="projectSection__links"
    />
    <summary
      className="projectSection__summary"
      onClick={[Function]}
    />
  </section>
  <footer
    className="projectSection__footer"
    data-active={true}
    onClick={[Function]}
  >
    <p
      data-active={true}
    >
      ^
    </p>
  </footer>
</article>
`;

exports[`Render ProjectSection with website link 1`] = `
<article
  className="projectSection"
>
  <header
    className="projectSection__header"
    onClick={[Function]}
  >
    <h2
      className="projectSection__name"
    />
  </header>
  <section
    className="projectSection__description"
    data-active={true}
  >
    <div
      className="projectSection__languages"
      onClick={[Function]}
    />
    <nav
      className="projectSection__links"
    >
      <a
        className="projectSection__website"
        href="Website!"
        rel="noopener"
        target="_blank"
      >
        Website
      </a>
    </nav>
    <summary
      className="projectSection__summary"
      onClick={[Function]}
    />
  </section>
  <footer
    className="projectSection__footer"
    data-active={true}
    onClick={[Function]}
  >
    <p
      data-active={true}
    >
      ^
    </p>
  </footer>
</article>
`;

exports[`Render ProjectSection without props 1`] = `
<article
  className="projectSection"
>
  <header
    className="projectSection__header"
    onClick={[Function]}
  >
    <h2
      className="projectSection__name"
    />
  </header>
  <section
    className="projectSection__description"
    data-active={true}
  >
    <div
      className="projectSection__languages"
      onClick={[Function]}
    />
    <nav
      className="projectSection__links"
    />
    <summary
      className="projectSection__summary"
      onClick={[Function]}
    />
  </section>
  <footer
    className="projectSection__footer"
    data-active={true}
    onClick={[Function]}
  >
    <p
      data-active={true}
    >
      ^
    </p>
  </footer>
</article>
`;
